reframe/core/modules.py

```diff
@@ -1030,6 +1031,7 @@ def available_modules(self, substr):
 
         return ret
 
+    @functools.lru_cache
```

Comment:
I think the problem is that the variant of the decorator without arguments was added in Python 3.8. So I suggest to change it to a backward compatible syntax:

```suggestion
    @functools.lru_cache(maxsize=128)
```

Reply:
I made this change